lib/cocoapods/target/build_settings.rb

```diff
@@ -960,9 +961,10 @@ def self.build_settings_names
         # @param [String] configuration_name
         #   see {#configuration_name}
         #
-        def initialize(target, configuration_name)
+        def initialize(target, configuration_name, configuration: nil)
           super(target)
           @configuration_name = configuration_name
+          (@configuration = configuration) || raise("No configuration for #{self}.")
```

Comment:
@segiddins why would we make this parameter optional if we are going to crash if its set to `nil`?

Reply:
because we can't have non-optional named parameters in ruby 2.0, they were introduced in 2.1